lib/matobj.gi

```diff
@@ -1568,3 +1568,394 @@ InstallMethod( \[\,\]\:\=, "for a matrix object, two positions, and an object",
   end );
 
 
+############################################################################
+# Elementary matrix operations
+############################################################################
+
+############################################################################
+##
+#O  MultMatrixRow( <M>, <row>, <scalar> )
+##
+##  <#GAPDoc Label="MultMatrixRow">
+##  <ManSection>
+##  <Oper Name="MultMatrixRow" Arg='mat,i,elm' Label=""/>
+##  <Oper Name="MultMatrixRowLeft" Arg='mat,i,elm' Label=""/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  Multiplies the <A>i</A>-th row of the mutable matrix <A>mat</A> with the scalar 
+##  <A>elm</A> from the left in-place.
+##  <P/>
+##  <C>MultMatrixRowLeft</C> is a synonym for <C>MultMatrixRow</C>.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( MultMatrixRow, "for a mutable matrix object, one row number, and a scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < row and row <= NrRows(mat) ) then
+        ErrorNoReturn("the second argument <row> has to fulfill 0 < row <= NrRows(mat) ");
```

Comment:
Tbh I don't know if it does. Actually in 26.7-1 the manual says "It is not specified what happens if i is larger than the length of v" for vectors, the entry for mattrix element access in 26.11-1 doesn't mention whether it does bound checks or not.

Reply:
Looking at the code for `IsPlistMatrixRep` it doesn't look like any bound checks are performed. Am I missing something?

```
InstallMethod( MatElm, "for a plist matrix and two positions",
  [ IsPlistMatrixRep, IsPosInt, IsPosInt ],
  function( m, row, col )
    return m![ROWSPOS][row]![ELSPOS][col];
  end );
```

Reply:
But that ends up performing a bounds check, because in GAP access to a list via `list[idx]` performs a bounds check.

Reply:
Ah alright. We just removed the bound checks. The only reason I still kind of like the idea of having them is the Error Message that's shown. When we remove them its a rather cryptic one that at least less experienced users might have trouble with to figure out what the actual problem was. That of course makes sense since the function that throws the error can't know in what context it was called. Our message on the other hand was quite clearly pointing at the wrong argument and told the user what range was expected.

Of course performance wise performing two bound checks doesn't make sense. So should we just remove them and live with a less clear error message or implement a NC version? I'm a bit torn what makes the most sense.
We'll remove it for now though. That solution to get a clearer error message is quite hacky and probably not the right way to do that.